### DISCUSSION TITLE
How does pocketbase auth deal with user logging in with different oauth2 provider but the same email?

### DISCUSSION THREAD

**Discussion by HuakunShen** [pocketbase/pocketbase#5500] [Q&A] [answered]
I see that when I login via GitHub, a user gets created in `users` table, but there is not provider column in the table. 

I see that there is a **Authorized providers** tab in edit record page, but I didn't find a way to query the provider. Given a logged in user, how to get its auth provider on client side and also on server side with Golang (given the auth token)?

If the user login with google with the same email, and same username what happens? What about same email but different username?

Is there going to be another row created, or one user gets multiple authorized providers, and the 2 logins merged into one account?

**Comment by ganigeorgiev** [comment, marked as answer]
I'm having difficulties understanding your questions, but for retrieving the linked OAuth2 accounts you can check https://pocketbase.io/docs/api-records/#list-linked-external-auth-providers.

> If the user login with google with the same email, and same username what happens? What about same email but different username?

The `username` doesn't matter. It is used only during the first PocketBase account creation. The linking between accounts is based on the `email` (or the `Authorization:TOKEN` header if set).

A single OAuth2 account cannot be linked to multiple PocketBase users.

If the above doesn't help, please try to elaborate more clearly what actually is the problem and your concerns and not just asking multiple random questions.

**Reply by HuakunShen** [reply]
Thank you!
So if a user login with both GitHub and Google using the same email, they eventually log in to the same account, right?

**Reply by ganigeorgiev** [reply]
Yes.
